ambry-router/src/main/java/com/github/ambry/router/TtlUpdateOperation.java

```diff
@@ -320,21 +331,17 @@ private void checkAndMaybeComplete() {
         operationException.set(
             new RouterException("TtlUpdateOperation failed because of BlobNotFound", RouterErrorCode.BlobDoesNotExist));
       }
+      if (quotaChargeCallback != null) {
+        try {
+          quotaChargeCallback.chargeQuota();
+        } catch (RouterException rEx) {
+          LOGGER.info("Exception {} while charging quota for ttl update operation.", rEx.toString());
+        }
+      }
```

Comment:
we are charging too many for TTLUPDATE and UNDELETE. 
We should only charge once for TTL_UPDATE and UNDELETE, and we already checked it on TtlUpdateManager and UndeleteManager, we don't need to change it again in TtlUpdateOperation.

Reply:
ttlupdate and undelete have different flows for a simple blob and for a composite blob. This charge here will happen only for the case of a simple blob's ttl update. The reason being that for the BatchOperation, the quotaChargeCallback is passed as null. Hence for composite blobs, only one charge will happen when all the blobs are updated.

Reply:
that makes sense